helix/machine_learning/models/KANs.py

```diff
@@ -0,0 +1,907 @@
+            lr=lr,
+            batch=batch,
+            problem_type=ProblemTypes.Regression,
+            loading_model=loading_model,
+        )
+
+    def predict(self, X):
+
+        X = self.sanitise_X(X)
+        y_pred = self.forward(X).detach().numpy().ravel()
+
+        return y_pred
+
+
+def f_inv(x, y_th):
+    x_th = 1 / y_th
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv2(x, y_th):
+    x_th = 1 / y_th ** (1 / 2)
+    return (
+        x_th,
+        y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**2) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv3(x, y_th):
+    x_th = 1 / y_th ** (1 / 3)
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**3) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv4(x, y_th):
+    x_th = 1 / y_th ** (1 / 4)
+    return (
+        x_th,
+        y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**4) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_inv5(x, y_th):
+    x_th = 1 / y_th ** (1 / 5)
+    return (
+        x_th,
+        y_th / x_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / x**5) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_sqrt(x, y_th):
+    x_th = 1 / y_th**2
+    return (
+        x_th,
+        x_th / y_th * x * (torch.abs(x) < x_th)
+        + torch.nan_to_num(torch.sqrt(torch.abs(x)) * torch.sign(x))
+        * (torch.abs(x) >= x_th),
+    )
+
+
+def f_power1d5(x, y_th):
+    return torch.abs(x) ** 1.5
+
+
+def f_invsqrt(x, y_th):
+    x_th = 1 / y_th**2
+    return (
+        x_th,
+        y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(1 / torch.sqrt(torch.abs(x))) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_log(x, y_th):
+    x_th = torch.e ** (-y_th)
+    return (
+        x_th,
+        -y_th * (torch.abs(x) < x_th)
+        + torch.nan_to_num(torch.log(torch.abs(x))) * (torch.abs(x) >= x_th),
+    )
+
+
+def f_tan(x, y_th):
+    clip = x % torch.pi
+    delta = torch.pi / 2 - torch.arctan(y_th)
+    return (
+        clip,
+        -y_th / delta * (clip - torch.pi / 2) * (torch.abs(clip - torch.pi / 2) < delta)
+        + torch.nan_to_num(torch.tan(clip)) * (torch.abs(clip - torch.pi / 2) >= delta),
+    )
+
+
+def f_arctanh(x, y_th):
+    delta = 1 - torch.tanh(y_th) + 1e-4
+    return (
+        delta,
+        y_th * torch.sign(x) * (torch.abs(x) > 1 - delta)
+        + torch.nan_to_num(torch.arctanh(x)) * (torch.abs(x) <= 1 - delta),
+    )
+
+
+def f_arcsin(x, y_th):
+    return (
+        (),
+        torch.pi / 2 * torch.sign(x) * (torch.abs(x) > 1)
+        + torch.nan_to_num(torch.arcsin(x)) * (torch.abs(x) <= 1),
+    )
+
+
+def f_arccos(x, y_th):
+    return (
+        (),
+        torch.pi / 2 * (1 - torch.sign(x)) * (torch.abs(x) > 1)
+        + torch.nan_to_num(torch.arccos(x)) * (torch.abs(x) <= 1),
+    )
+
+
+def f_exp(x, y_th):
```

Comment:
```suggestion
def _f_exp(x, y_th):
```
Leading underscore to indicate this is private to this module. Change it in the dict below too